apis/v1alpha1/httproute_types.go

```diff
@@ -532,6 +546,45 @@ type HTTPRequestMirrorFilter struct {
 	Port PortNumber `json:"port"`
 }
 
+// HTTPRequestAuthorization defines configuration for the RequestAuthorization filter.
+type HTTPRequestAuthorization struct {
+	// URL indicates the address of the service used to perform authorization checks.
+	//
+	// There are two protocols supported:
+	// - grpc: In this mode, the authorization server should receive a grpc request of type <service.auth.v3.CheckRequest>.
+	//   It is expected for the authorization server to implement the Envoy ExtAuthz GRPC protocol:
+	//   https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto
+	//
+	// - https: An HTTP request that will include all the Headers and Request Path, but no body content.
+	//   It is expected for the authorization server to implement the Envoy ExtAuthz HTTP protocol:
+	//   https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto
+	//
+	URL string `json:"url"`
+
+	// FailModeAllow defines how this authorization filter should be handled if it timeouts or fails to
+	// contact the URL defined. This field accepts two options:
+	//
+	// - True: Fail open. Traffic will be allowed even if URL can't be contacted or timeouts
+	// - False: Fail closed. Default value. Traffic will be denied if the gateway can't reach the URL or timeouts.
+	//
+	// +optional
+	// +kubebuilder:default=false
+	FailModeAllow *bool `json:"failModeAllow,omitempty"`
+
+	// TimeoutMs is the duration that the authorization call is allowed to operate.
+	// Default if unset is 100
+	//
+	// +optional
+	// +kubebuilder:default= 100
+	TimeoutMs int `json:"timeoutMs,omitempty"`
+
+	// Metadata can be used to provide additional context information for the authorization
+	// endpoint to make a decision.
+	//
+	// +optional
+	Metadata *map[string]string `json:"metadata,omitempty"`
```

Comment:
a pointer to a map feels a bit odd, is this used in other parts of the API?

Reply:
That doesn't look ok, I've changed this type from `string` to a `map[string]string` and left it there. Let me fix it. Thanks.